### PR TITLE
Document order patch 1

### PR DESCRIPTION
**Pull request by JonidBendo**
- Add Event after the positions are processed for the invoice generation
- Add set and get methods for the protected variables to allow further modifications for taxes and etc.

## Description
Please describe your pull request:
* Why is it necessary?
There is no other way to actually work/edit the invoice generation system

* What does it improve?
It allows plugin developers to use the event and modify the invoice details

* Does it have side effects?
If not used properly from the plugin developers it might break the invoice generation methods

| Questions        | Answers
| ---------------- | -------------------------------------------------------
| BC breaks?       | no
| Tests pass?      | yes
| Related tickets? | If this PR fixes an existing [Issue](https://issues.shopware.com) ticket, please add its complete issue URL.
| How to test?     | Simply call the event in a plugin...

### CODE REVIEW

File: engine/Shopware/Models/Document/Order.php [hunk 1]
```diff
@@ -527,9 +527,48 @@ public function processPositions()
 
             $this->_positions->offsetSet($key, $position);
         }
+        
+        $parameters = $this->getParameters();
+        $parameters = Shopware()->Events()->filter('DOCUMENT_ORDER_POSITIONS_PROCESSED', $parameters, array('subject' => $this, 'params' => $parameters));
```
Comment (nlubisch) [inline engine/Shopware/Models/Document/Order.php]: use short array syntax and chop the line down to improve the readability

File: engine/Shopware/Models/Document/Order.php [hunk 2]
```diff
@@ -527,9 +527,48 @@ public function processPositions()
...
+        $this->setParameters($parameters);
+    }
+   
+    /**
+     * Get class parameters
+     * @return Array $parameters
+     */
+    private function getParameters()
+    {
+        $parameters['positions']    = $this->_positions;
```
Comment (nlubisch) [inline engine/Shopware/Models/Document/Order.php]: remove the identation before the values to provide a consistent code style

File: engine/Shopware/Models/Document/Order.php [hunk 3]
```diff
@@ -527,9 +527,48 @@ public function processPositions()
...
+        $parameters['tax']          = $this->_tax;
+        $parameters['net']          = $this->_net;
+        $parameters['shipping']     = $this->_shipping;
+        $parameters['user']         = $this->_user;
+
+        return $parameters;
     }
 
     /**
+     * Se class parameters
```
Comment (nlubisch) [inline engine/Shopware/Models/Document/Order.php]: typo

File: engine/Shopware/Models/Document/Order.php [hunk 4]
```diff
@@ -527,9 +527,48 @@ public function processPositions()
...
+     * @param  $parameters
+     * @return void
+     */
+    private function setParameters( $parameters )
```
Comment (nlubisch) [inline engine/Shopware/Models/Document/Order.php]: remove space before and after parameter

**Review by nlubisch — CHANGES_REQUESTED** [review]

File: engine/Shopware/Models/Document/Order.php [hunk 5]
```diff
@@ -527,9 +527,48 @@ public function processPositions()
...
         }
+        
+        $parameters = $this->getParameters();
+        $parameters = Shopware()->Events()->filter('DOCUMENT_ORDER_POSITIONS_PROCESSED', $parameters, array('subject' => $this, 'params' => $parameters));
+        $this->setParameters($parameters);
+    }
+   
+    /**
+     * Get class parameters
+     * @return Array $parameters
```
Comment (nlubisch) [inline engine/Shopware/Models/Document/Order.php]: return type to lower case

**Comment by OliverSkroblin** [comment]
Hi @JonidBendo,
thanks for your pull request.

We merged your changes. The changes will be available in version 5.3.5

**Pull request closed without merging**